specification.md

```diff
@@ -301,6 +332,10 @@ explicit runtime type-checks, but it is not required to do so.
 - Getters/setters can be replaced by properties, depending on what is customary 
 in the language in which the library is implemented.
 
+- Implementations may elect, at their discretion, to implement many of the 
+classes described in this document as read-only classes for which all 
+properties are set at construction time. 
+
```

Comment:
It is an academic argument as long as we have only one implementation, but the statement above would limit application portability: an application using setters in one implementation could not easily switch to another implementation which has the proposed read-only semantics.

Reply:
Good point. And Python is rather unique in commonly having constructors with n >> 1 arguments due to kwargs. So perhaps it might be better to leave this out.

Reply:
Removed